docs/content/manual/_incoming/2175-2332-harvester-pull-rancher-image-private-registry.md

```diff
@@ -0,0 +1,112 @@
+---
+title: Harvester pull Rancher agent image from private registry
+---
+
+* Related issues: [#2175](https://github.com/harvester/harvester/issues/2175) [BUG] Harvester fails to pull Rancher agent image from private registry
+* Related issues: [#2332](https://github.com/harvester/harvester/issues/2332) [Backport v1.0] Harvester fails to pull Rancher agent image from private registry
+
+## Category: 
+* Virtual Machine
+
+## Verification Steps
+1. Create a harvester cluster and a ubuntu server. Make sure they can reach each other.
+1. On the ubuntu server, install docker and run the following commands.
+```
+mkdir -p certs
+openssl req \
+  -newkey rsa:4096 -nodes -sha256 -keyout certs/domain.key \
+  -addext "subjectAltName = DNS:myregistry.local" \
+  -x509 -days 365 -out certs/domain.crt
+sudo mkdir -p /etc/docker/certs.d/myregistry.local:5000
+sudo cp certs/domain.crt /etc/docker/certs.d/myregistry.local:5000/domain.crt
+docker run -d \
+  -p 5000:5000 \
+  --restart=always \
+  --name registry \
+  -v "$(pwd)"/certs:/certs \
+  -v "$(pwd)"/registry:/var/lib/registry \
+  -e REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt \
+  -e REGISTRY_HTTP_TLS_KEY=/certs/domain.key \
+  registry:2
+docker pull rancher/rancher-agent:v2.6.5
+docker tag rancher/rancher-agent:v2.6.5 myregistry.local:5000/rancher/rancher-agent:v2.6.5
+docker push myregistry.local:5000/rancher/rancher-agent:v2.6.5
+```
```

Comment:
Will this work? I thought that for a private registry that was on a different server from the Rancher server you needed to add basic HTTP auth and then use `docker login myregistry.local:5000`.

Reply:
I install a new Ubuntu VM and follow the steps to double confirm. 
If I follow the original steps. The last push docker image would failed since no such host found.  
```
davidtclin@davidtclin-Standard-PC-Q35-ICH9-2009:~$ sudo docker push myregistry.local:5000/rancher/rancher-agent:v2.6.5
The push refers to repository [myregistry.local:5000/rancher/rancher-agent]
Get "https://myregistry.local:5000/v2/": dial tcp: lookup myregistry.local: no such host
```

The correct way should move the following step 3 prior to running docker registry command  
* On each harvester node, add ubuntu IP to `/etc/hosts`.
  ```
  # vim /etc/hosts
  <host ip> myregistry.local
  ```

Then run the docker push command can work
```
davidtclin@davidtclin-Standard-PC-Q35-ICH9-2009:~$ sudo docker push myregistry.local:5000/rancher/rancher-agent:v2.6.5
The push refers to repository [myregistry.local:5000/rancher/rancher-agent]
2005654ae76b: Pushed 
0c19c204d107: Pushed 
c1135b4a3bc3: Pushed 
a7abce8c9d73: Pushed 
800376671444: Pushed 
8d44ad5ec4d0: Pushed 
f36bdc4193db: Pushed 
e0d05f5a091a: Pushed 
v2.6.5: digest: sha256:eb4c1eaae3bd0c0881874f65f20a847a40de75b5a3399d2c7979cb0e78a3c827 size: 2005
```

Thanks for the check. I have moved the step 2 prior to step 3 and outline each command for better readibility.